src/native/libs/System.Security.Cryptography.Native.Apple/CMakeLists.txt

```
@@ -42,16 +42,38 @@ endif()
 # As of CMake 3.20.2 support for Swift only works with the Ninja and XCode
 # generators so we cannot rely on it. Even with the Ninja generator it doesn't
 # work in combination with other languages within the same library.
-if (NOT SWIFT_COMPILER_TARGET AND CLR_CMAKE_TARGET_OSX)
-    set(SWIFT_PLATFORM "macosx")
-    set(SWIFT_PLATFORM_SUFFIX "")
+if (NOT SWIFT_COMPILER_TARGET)
     set(SWIFT_DEPLOYMENT_TARGET ${CMAKE_OSX_DEPLOYMENT_TARGET})
+    set(SWIFT_PLATFORM_SUFFIX "")
+
+    if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(SWIFT_OPTIMIZATION_FLAG "-Onone")
+    elseif (CMAKE_BUILD_TYPE STREQUAL "Release")
+        set(SWIFT_OPTIMIZATION_FLAG "-O")
+    endif()
+
+    if (CLR_CMAKE_TARGET_OSX)
+        set(SWIFT_PLATFORM "macosx")
+    elseif (CLR_CMAKE_TARGET_MACCATALYST)
+        set(SWIFT_PLATFORM "ios")
+        set(SWIFT_PLATFORM_SUFFIX "-macabi")
+    elseif (CLR_CMAKE_TARGET_IOS)
+        set(SWIFT_PLATFORM "ios")
+        if(CMAKE_OSX_SYSROOT MATCHES iPhoneSimulator)
```

```suggestion
        if (CMAKE_OSX_SYSROOT MATCHES iPhoneSimulator)
```

All the others ifs have a space after them.